vpr/src/route/rr_graph_generation/interposer_cut.cpp

```diff
@@ -0,0 +1,266 @@
+
+            if (should_cut(src_start_loc_y, sink_start_loc_y, cut_loc_y)) {
+                edges_to_be_removed.push_back(edge_id);
+            }
+        }
+
+        for (int cut_loc_x : grid.get_vertical_interposer_cuts()[layer]) {
+            int src_start_loc_x = node_xstart(rr_graph, src_node);
+            int sink_start_loc_x = node_xstart(rr_graph, sink_node);
+
+            if (should_cut(src_start_loc_x, sink_start_loc_x, cut_loc_x)) {
+                edges_to_be_removed.push_back(edge_id);
+            }
+        }
+    }
+
+    return edges_to_be_removed;
+}
+
+/**
+  * @brief Update a CHANY node's bounding box in RRGraph and SpatialLookup entries.
+  * This function assumes that the channel node actually crosses the cut location and
+  * might not function correctly otherwise.
+  *
+  * This is a low level function, you should use cut_channel_node that wraps this up in a nicer API.
+  */
+static void cut_chan_y_node(RRNodeId node, int x_low, int y_low, int x_high, int y_high, int layer, int ptc_num, int cut_loc_y, Direction node_direction, RRGraphBuilder& rr_graph_builder, RRSpatialLookup& spatial_lookup) {
+    if (node_direction == Direction::INC) {
+        // Anything above cut_loc_y shouldn't exist
+        rr_graph_builder.set_node_coordinates(node, x_low, y_low, x_high, cut_loc_y);
+
+        // Do a loop from cut_loc_y to y_high and remove node from spatial lookup
+        for (int y_loc = cut_loc_y + 1; y_loc <= y_high; y_loc++) {
+            spatial_lookup.remove_node(node, layer, x_low, y_loc, e_rr_type::CHANY, ptc_num);
+        }
+    } else if (node_direction == Direction::DEC) {
+        // Anything below cut_loc_y (inclusive) shouldn't exist
+        rr_graph_builder.set_node_coordinates(node, x_low, cut_loc_y + 1, x_high, y_high);
+
+        // Do a loop from y_low to cut_loc_y and remove node from spatial lookup
+        for (int y_loc = y_low; y_loc <= cut_loc_y; y_loc++) {
+            spatial_lookup.remove_node(node, layer, x_low, y_loc, e_rr_type::CHANY, ptc_num);
+        }
+    } else {
+        VTR_ASSERT_MSG(false, "Bidirectional routing is not supported for interposer architectures.");
+    }
+}
+
+/**
+  * @brief Update a CHANX node's bounding box in RRGraph and SpatialLookup entries.
+  * This function assumes that the channel node actually crosses the cut location and
+  * might not function correctly otherwise.
+  *
+  * This is a low level function, you should use cut_channel_node that wraps this up in a nicer API.
+  */
+static void cut_chan_x_node(RRNodeId node, int x_low, int y_low, int x_high, int y_high, int layer, int ptc_num, int cut_loc_x, Direction node_direction, RRGraphBuilder& rr_graph_builder, RRSpatialLookup& spatial_lookup) {
+    if (node_direction == Direction::INC) {
+        // Anything to the right of cut_loc_x shouldn't exist
+        rr_graph_builder.set_node_coordinates(node, x_low, y_low, cut_loc_x, y_high);
+
```

Comment:
Add assert that the node crosses the cut location (or just return without doing anything and update the comment).